src/app/workflow/workflow.component.ts

```diff
@@ -398,15 +398,8 @@ export class WorkflowComponent extends Entry<WorkflowVersion> implements AfterVi
           this.updateWorkflowUrl(this.workflow);
         },
         (error) => {
-          const workflowOrgRegex = /\/workflows\/.+/;
-          const workflowRegex = /\/workflows\/(github.com)|(gitlab.com)|(bitbucket.org)\/.+/;
-          const gitHubAppToolRegex = /\/containers\/(github.com)\/.+/;
-          if (
-            workflowOrgRegex.test(this.resourcePath) ||
-            workflowRegex.test(this.resourcePath) ||
-            gitHubAppToolRegex.test(this.resourcePath)
-          ) {
-            this.router.navigate(['page-not-found']);
+          if (!this.workflow) {
+            this.router.navigate(['page-not-found'], { skipLocationChange: true });
```

Comment:
Thought to replace the regex condition to allow invalid workflow paths like `services/asdf`, `notebooks/asdf`, `apptools/asdf`, `search-workflows/asdf` to also display `PageNotFound` instead of a blank page (even though some of these URLs are unlikely to be encountered).

Reply:
Without testing it out, my initial thought is I'm usually in favour of avoiding a regex if possible. They can be hard to debug